app/import.go

```diff
@@ -169,18 +199,8 @@ func (a *App) bulkImport(c *request.Context, jsonlReader io.Reader, attachmentsR
 			return model.NewAppError("BulkImport", "app.import.bulk_import.json_decode.error", nil, err.Error(), http.StatusBadRequest), lineNumber
 		}
 
-		if len(attachedFiles) > 0 && line.Post != nil && line.Post.Attachments != nil {
-			for i, attachment := range *line.Post.Attachments {
-				var ok bool
-				path := *attachment.Path
-				if (*line.Post.Attachments)[i].Data, ok = attachedFiles[path]; !ok {
-					return model.NewAppError("BulkImport", "app.import.bulk_import.json_decode.error", nil, fmt.Sprintf("attachment '%s' not found in map", path), http.StatusBadRequest), lineNumber
-				}
-			}
-		}
-
-		if importPath != "" {
-			rewriteFilePaths(&line, importPath)
+		if err := processAttachments(&line, importPath, attachedFiles); err != nil {
+			return model.NewAppError("BulkImport", "app.import.bulk_import.process_attachments.error", nil, err.Error(), http.StatusBadRequest), lineNumber
```

Comment:
So I think here, we actually don't want to return `err`, we just want to log out the error, so that if for some reason attachments can't be added to a single post the import process continues on to other posts in the backup. This is because imports need to be a best-effort attempt to import as much of the data as possible, because the automated importer can't stop to reconcile problems encountered midway, and it's better to import as much as possible.

Reply:
This is just about path/file matching though. We were returning before as well so it's not a change. This is prior to any uploading, just validating the input data essentially. We are still allowing for uploads to fail as per your recent PR.